django2/files/.platform.app.yaml

```diff
@@ -0,0 +1,62 @@
+# This file describes an application. You can have multiple applications
+# in the same project.
+#
+# See https://docs.platform.sh/user_guide/reference/platform-app-yaml.html
+
+# The name of this app. Must be unique within a project.
+name: 'app'
+
+# The runtime the application uses.
+type: 'python:3.7'
+
+# The build-time dependencies of the app.
+dependencies:
+    python3:
+       pipenv: '2018.10.13'
+
+# The relationships of the application with services or other applications.
+#
+# The left-hand side is the name of the relationship as it will be exposed
+# to the application in the PLATFORM_RELATIONSHIPS variable. The right-hand
+# side is in the form `<service name>:<endpoint name>`.
+relationships:
+    database: "postgresqldb:postgresql"
+
+# The configuration of app when it is exposed to the web.
+web:
+    # Whether your app should speak to the webserver via TCP or Unix socket
+    # https://docs.platform.sh/configuration/app-containers.html#upstream
+    upstream:
+        socket_family: unix
```

Comment:
I'm surprised by this.  All of the other examples we have use a port, not a unix socket.  Is there a reason for it?  Is this in the current example repo this way?

Reply:
I took this as a working code snippet from Django 1.X example without special inspection. It seems using `-b 0.0.0.0:$PORT` as a gunicorn flag works equally well. [Official Django docs page](https://docs.djangoproject.com/en/2.1/howto/deployment/wsgi/gunicorn/) also doesn't mention anything about unix sockets. Maybe we can ask original author @JGrubb if it can be simply replaced by `gunicorn -w 4 -b 0.0.0.0:$PORT myapp.wsgi:application`.

Reply:
Funny enough, I just started a Django project last night so I'll try using a port instead.

@Crell to your point though - we likely didn't have any/enough documentation on using sockets vs ports at the time this was put together.  Most of the Python and Ruby examples used to tend toward sockets instead of ports for whatever reason.

Reply:
If a port works equally well, I'd marginally favor switching to that for consistency's sake.  I always had trouble setting up socket listeners when making templates, but I suspect it was just a question of who built the template originally.  (The Ruby examples are mostly Ori, for instance.)